internal/cli/template_delete.go

```diff
@@ -0,0 +1,116 @@
+package cli
+
+import (
+	"github.com/posener/complete"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+type ProjectTemplateDeleteCommand struct {
+	*baseCommand
+
+	flagName string
+	flagID   string
+}
+
+func (c *ProjectTemplateDeleteCommand) Run(args []string) int {
+	flagSet := c.Flags()
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(flagSet),
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+	args = flagSet.Args()
+	ctx := c.Ctx
+
+	name := ""
+	if len(args) == 1 {
+		name = args[0]
+	}
+
+	var tref pb.Ref_ProjectTemplate
+	if name != "" {
+		tref.Ref = &pb.Ref_ProjectTemplate_Name{
+			Name: name,
+		}
+	} else if c.flagID != "" {
+		tref.Ref = &pb.Ref_ProjectTemplate_Id{
+			Id: c.flagID,
+		}
+		name = c.flagID
+	} else if c.flagName != "" {
+		tref.Ref = &pb.Ref_ProjectTemplate_Name{
+			Name: c.flagName,
+		}
+		name = c.flagName
+	} else {
+		c.ui.Output("Missing project template name or id.\n\n"+c.Help(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	_, err := c.project.Client().DeleteProjectTemplate(ctx, &pb.DeleteProjectTemplateRequest{
+		ProjectTemplate: &tref,
+	})
+	if err != nil {
+		c.ui.Output("Encountered an error while deleting the project template: %s", clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+
+	c.ui.Output("Template %q deleted", name, terminal.WithSuccessStyle())
```

Comment:
nit: both `create` and `update` report success with a lowercase `t` for template

```suggestion
	c.ui.Output("template %q deleted", name, terminal.WithSuccessStyle())
```